### PR TITLE
Add support for influxdb data source

### PR DESCRIPTION
**Pull request by rochaporto**

### CODE REVIEW

File: internal/service/metric/influxdb/influxdb.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+	_ "github.com/influxdata/influxdb1-client" // needed due to go mod bug
+	influxdbv2 "github.com/influxdata/influxdb1-client/v2"
+
+	"github.com/slok/grafterm/internal/model"
+	"github.com/slok/grafterm/internal/service/metric"
+)
+
+// ConfigGatherer is the configuration of the InfluxDB gatherer.
+type ConfigGatherer struct {
```
Comment (slok) [inline internal/service/metric/influxdb/influxdb.go]: What do you think about injecting the HTTP client and defaulting to Go's STD library when the client is not passed?

This way the client with the password, TLS, username... settings would be on the upper layer, we could inject any kind of configured client and the Influx gatherer doesn't need to know anything about clients except the address to connect to.

e.g.

https://github.com/slok/grafterm/blob/3b4bb0f580613a54023bab34f2672c4579ba7f9c/internal/service/metric/graphite/graphite.go#L19-L26

**Comment by rochaporto** [comment]
> Hi @rochaporto
> 
> Thanks for your contribution, this is **awesome**! 
> 
> I don't have an influx to test myself, I have never used before.
> 
> I put you some comments on the PR, what do you think about it?
> 
> One last thing :) do you know any public influxdb so we can create a dashboard example so users can use as an example for their own influxdb dashboards? (would be on a different PR).

I'm not aware of one, just gave it a search as well. I have a test one which is in fact what i use against our production metrics db but that won't help. If i find something i'll gladly push an example dashboard.

File: internal/service/metric/influxdb/influxdb.go (hunk 1 above)
Comment (rochaporto) [reply internal/service/metric/influxdb/influxdb.go]: Sounds great, i'll look at this. The first version was following the graphite example but i changed it to help with the tests. It now makes more sense why graphite had that.

Comment (rochaporto) [reply internal/service/metric/influxdb/influxdb.go]: I couldn't find an easy way to do this (injecting the std http client). The library's implementation seems to be taking over the role: https://github.com/influxdata/influxdb1-client/blob/master/v2/client.go

Maybe i'm missing something?

**Comment by slok** [comment]
No problem!thanks againt!

File: internal/service/metric/influxdb/influxdb.go (hunk 1 above)
Comment (slok) [reply internal/service/metric/influxdb/influxdb.go]: Sorry, you are right! I meant the influx default client, in this case, an influx client and not an STD client.

Comment (rochaporto) [reply internal/service/metric/influxdb/influxdb.go]: Let me know if this is what you meant, just pushed an update.

Comment (slok) [reply internal/service/metric/influxdb/influxdb.go]: Yes, that's it! could you make a validation that the configuration data received in `ConfigGatherer` is ok? for example, if no addr error, if not client create a default one... whatever you think is better for each of the config paramaters.

Comment (rochaporto) [reply internal/service/metric/influxdb/influxdb.go]: Default client added, i believe the influxdb client will already do the addr validation and return an appropriate error. Database validation also added so it's mandatory.

File: internal/service/metric/influxdb/influxdb.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+type gatherer struct {
+	cli influxdbv2.Client
+	cfg ConfigGatherer
+}
+
+// NewGatherer returns a new metric gatherer for influxdb backends.
+func NewGatherer(cfg ConfigGatherer) (metric.Gatherer, error) {
+	err := cfg.defaults()
+	if err != nil {
+		return &gatherer{}, nil
```
Comment (slok) [inline internal/service/metric/influxdb/influxdb.go]:
```suggestion
		return &gatherer{}, err
```

File: internal/service/metric/influxdb/influxdb.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	}
+
+	if cfg.Database == "" {
```
Comment (slok) [inline internal/service/metric/influxdb/influxdb.go]: Could you move this to the `defaults` so everything is defaulted and checked in the same place? thanks.

**Comment by slok** [comment]
Thank you for all the effort and the code! and sorry for all the requested changes. When I have time some spear time I'll try creating an influx dashboard and publish a new release.

Thanks once again!

**Pull request merged**